src/proto/streams/state.rs

```diff
@@ -74,6 +74,11 @@ enum Peer {
 enum Cause {
     Proto(Reason),
     Io,
+
+    /// The user droped all handles to the stream without explicitly canceling.
+    /// This indicates to the connection that a reset frame must be sent out
+    /// once the send queue has been flushed.
+    Canceled,
```

Comment:
I think there's a few blanket matches that will match this case, like in `ensure_recv_open` down below...

Reply:
I fixed `ensure_recv_open ` but that is the only one that seemed problematic?